packages/dd-trace/src/plugins/util/git.js

```diff
@@ -47,16 +48,20 @@ function getCommitsToUpload (commitsToExclude) {
 function generatePackFilesForCommits (commitsToUpload) {
   const tmpFolder = os.tmpdir()
 
-  const prefix = Math.floor(Math.random() * 10000)
-  const path = `${tmpFolder}/${prefix}`
+  const randomPrefix = Math.floor(Math.random() * 10000)
+  const temporaryPath = path.join(tmpFolder, randomPrefix)
 
-  const orderedCommits =
-    execSync(
-      `git pack-objects --compression=9 --max-pack-size=3m ${path}`,
-      { input: commitsToUpload.join('\n') }
-    ).toString().split('\n').filter(commit => !!commit)
+  try {
+    const orderedCommits =
+      execSync(
+        `git pack-objects --compression=9 --max-pack-size=3m ${temporaryPath}`,
+        { input: commitsToUpload.join('\n') }
+      ).toString().split('\n').filter(commit => !!commit)
 
-  return orderedCommits.map(commit => `${path}-${commit}.pack`)
+    return orderedCommits.map(commit => `${temporaryPath}-${commit}.pack`)
+  } catch (e) {
```

Comment:
Should we at least log the error here? Otherwise when generating packfiles fail there would be no way to understand why.

Reply:
the empty state is an error state (this should always generate at least a pack file), and we record it here: https://github.com/DataDog/dd-trace-js/pull/2408/files#diff-c1f72af0e283f5ff39cb79aff7209bd24d2ed72ad1b92b3241d61398df77bf5dR190

Reply:
That only logs a new error though, not the original error that explains _why_ it failed.